newdle/api.py

```diff
@@ -200,6 +200,17 @@ def get_my_newdles():
     return MyNewdleSchema(many=True).jsonify(newdle)
 
 
+@api.route('/newdles/in')
+def get_newdles_im_in():
+    newdle = (
+        Newdle.query.options(noload('participants'))
+        .filter(Newdle.participants.any(auth_uid=g.user['uid']))
+        .order_by(Newdle.final_dt.isnot(None), Newdle.final_dt.desc(), Newdle.id.desc())
```

Comment:
I'm assuming a newdle someone has been invited to, but didn't reply, should reflect a participant entry without answers (an empty JSON object {}).

The most naive approach would be to add a sql filter when the "answers" column is '{}' (which also includes converting the field to public). The alternative is to do a post filter.

Do you have a preference or any other suggestions?

Reply:
Checking for an empty object should work, and it's indeed the way how to check if no answers have been submitted yet.

The reason why it's `_answers` (assuming that's what you mean with it not being public right now) is that we needed the property to convert to a nicer format in the Python side.

If you make it an `@hybrid_property` (from sqlalchemy) instead of `@property`, you can add an `@answers.expression` method (classmethod-ish, i.e. `cls` as the first arg) which just returns `cls._answers`. That way you can use `Participant.answers` when constructing the query.

Reply:
Yes, I figured we probably wanted to keep the raw `_answers` private.
Thanks, I never knew about sqlalchemy's @hybrid_property annotation. I will add that.